Source/Csla.Generators/readme.md

```diff
@@ -0,0 +1,97 @@
+# Using the Source Generator with AddressPOCO in CSLA.NET
+
+This guide provides a step-by-step approach to using the source generator in the CSLA.NET project, specifically with the `AddressPOCO.cs` class as an example. The source generator is a powerful tool that can automatically generate additional source code for your projects, enhancing productivity and maintainability.
+
+## Prerequisites
+
+- Visual Studio 2019 or later with the .NET Core development workload installed.
+- The CSLA.NET NuGet package added to your project.
+- Ths CSLA.NET Source Generator NuGet package added to your project.
+
+## Step 1: Define Your POCO Class
+
+Start by defining your Plain Old CLR Object (POCO) class. In this example, we use `AddressPOCO.cs`:
+
+```csharp
+using Csla.Serialization;
+
+namespace Csla.Generators.CSharp.TestObjects
+{
+
+  [AutoSerializable]
+  public partial class AddressPOCO
+  {
+
+    public string? AddressLine1 { get; set; }
+
+    public string AddressLine2 { get; set; } = string.Empty;
+
+    public string Town { get; set; } = string.Empty;
+
+    public string County { get; set; } = string.Empty;
+
+    public string Postcode { get; set; } = string.Empty;
+
+  }
+}
+```
+
+## Understanding [AutoSerializable]
+
+- **Source Generators:** Introduced in .NET 5, source generators allow for compile-time code generation based on the existing codebase. They are a powerful tool for reducing boilerplate code and automating repetitive coding tasks.
+- **[AutoSerializable] Attribute:** This custom attribute is used to mark a class for automatic serialization code generation. When the source generator sees this attribute, it generates the necessary serialization and deserialization methods for the class, tailored to its properties.
```

Comment:
I think the `AutoSerializable` class is in the `Csla` project, but really that class should be in the generators project. Same namespace, but the file/type should only be available if the user has chosen to use this particular serializer by referencing that NuGet package.

Reply:
I didn't understand what I need to change, Do you want the autoSerializable to be moved to the Generator Project?

Reply:
Yes, exactly. It is in the csla project now, but should be in the generator project.

Reply:
Moved to the generator. there are other 2 unused attributes, we could talk about them later